base/sparse/higherorderfns.jl

```diff
@@ -871,11 +874,11 @@ promote_containertype(::Type{AbstractSparseArray}, ::Type{Tuple}) = Array
 
 # broadcast[!] entry points for combinations of sparse arrays and other (scalar) types
 @inline function broadcast_c{N}(f, ::Type{AbstractSparseArray}, mixedargs::Vararg{Any,N})
-    parevalf, passedargstup = capturescalars(f, mixedargs)
+    parevalf, passedargstup = capturescalars(f, map(_sparsifystructured, mixedargs))
```

Comment:
Given the `capturescalars` pathway's standing shortcomings, sending sparse `broadcast[!]` calls through `capturscalars` that need not go through `capturscalars` makes me uneasy. (Overcoming `capturescalars`'s shortcomings would be wonderful. Any thoughts on that front?)